Create a C++ implementation of WebExtensionSQLite classes

ItsJamie9494 opened the pull request:
#### 0d58f6b2b12f66664bdc7e052869a8d406967b21
<pre>
Create a C++ implementation of WebExtensionSQLite classes
<a href="https://webkit.org/b/285893">https://webkit.org/b/285893</a>

Reviewed by Timothy Hatcher.

This adds a new WebExtensionSQLite implementation, with the Database, Helpers, Row, Statements, and Store classes.
Currently, there is no usage of this implementation, as StorageSQLiteStore, RegisteredScriptsSQLiteStore, and DeclarativeNetRequestSQLiteStore should be ported separately.

* Source/WebKit/Scripts/generate-unified-sources.sh:
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.cpp: Added.
(WebExtensionSQLiteDatabase::WebExtensionSQLiteDatabase):
(WebExtensionSQLiteDatabase::assertQueue):
(WebExtensionSQLiteDatabase::close):
(WebExtensionSQLiteDatabase::reportErrorWithCode):
(WebExtensionSQLiteDatabase::errorWithSQLiteErrorCode):
(WebExtensionSQLiteDatabase::enableWAL):
(WebExtensionSQLiteDatabase::openWithAccessType):
(WebExtensionSQLiteDatabase::inMemoryDatabaseURL):
(WebExtensionSQLiteDatabase::privateOnDiskDatabaseURL):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.h: Added.
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatatypeTraits.h: Added.
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;int&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;int&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;int64_t&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;int64_t&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;double&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;double&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;String&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;String&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;RefPtr&lt;API::Data&gt;&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;RefPtr&lt;API::Data&gt;&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;std::nullptr_t&gt;::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;std::nullptr_t&gt;::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits&lt;decltype):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteHelpers.h: Added.
(WebKit::SQLiteIsExecutionError):
(WebKit::SQLiteDatabaseExecuteAndReturnIntError):
(WebKit::SQLiteStatementBindAllParameters):
(WebKit::SQLiteDatabaseExecute):
(WebKit::SQLiteDatabaseExecuteAndReturnError):
(WebKit::SQLiteDatabaseFetch):
(WebKit::SQLiteStatementBindParameter):
(WebKit::SQLiteStatementBindParameter&lt;unsigned&gt;):
(WebKit::SQLiteStatementBindParameter&lt;uint64_t&gt;):
(WebKit::SQLiteStatementBindParameter&lt;long&gt;):
(WebKit::SQLiteStatementBindParameter&lt;bool&gt;):
(WebKit::SQLiteStatementBindTupleParameters):
(WebKit::SQLiteIteratorBlock::SQLiteIteratorBlock):
(WebKit::SQLiteIteratorBlock::callBlockWithAllColumns):
(WebKit::StatementCallBlockWithAllColumns):
(WebKit::WBSStatementFetchColumnsInTuple):
(WebKit::SQLiteStatementBindOrStep):
(WebKit::SQLiteStatementBind):
(WebKit::SQLiteDatabaseEnumerate):
(WebKit::SQLiteDatabaseEnumerateRows):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.cpp: Added.
(WebKit::WebExtensionSQLiteRow::WebExtensionSQLiteRow):
(WebKit::WebExtensionSQLiteRow::getString):
(WebKit::WebExtensionSQLiteRow::getInt):
(WebKit::WebExtensionSQLiteRow::getInt64):
(WebKit::WebExtensionSQLiteRow::getDouble):
(WebKit::WebExtensionSQLiteRow::getBool):
(WebKit::WebExtensionSQLiteRow::getData):
(WebKit::WebExtensionSQLiteRow::isNullAtIndex):
(WebKit::WebExtensionSQLiteRowEnumerator::WebExtensionSQLiteRowEnumerator):
(WebKit::WebExtensionSQLiteRowEnumerator::next):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.h: Added.
(WebKit::WebExtensionSQLiteRow::create):
(WebKit::WebExtensionSQLiteRowEnumerator::create):
(WebKit::WebExtensionSQLiteRowEnumerator::statement):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStatement.cpp: Added.
(WebKit::WebExtensionSQLiteStatement::WebExtensionSQLiteStatement):
(WebKit::WebExtensionSQLiteStatement::~WebExtensionSQLiteStatement):
(WebKit::WebExtensionSQLiteStatement::execute):
(WebKit::WebExtensionSQLiteStatement::fetch):
(WebKit::WebExtensionSQLiteStatement::fetchWithEnumerationCallback):
(WebKit::WebExtensionSQLiteStatement::reset):
(WebKit::WebExtensionSQLiteStatement::invalidate):
(WebKit::WebExtensionSQLiteStatement::bind):
(WebKit::WebExtensionSQLiteStatement::columnNamesToIndicies):
(WebKit::WebExtensionSQLiteStatement::columnNames):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStatement.h: Added.
(WebKit::WebExtensionSQLiteStatement::create):
(WebKit::WebExtensionSQLiteStatement::database):
(WebKit::WebExtensionSQLiteStatement::handle):
(WebKit::WebExtensionSQLiteStatement::isValid):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStore.cpp: Added.
(WebKit::WebExtensionSQLiteStore::WebExtensionSQLiteStore):
(WebKit::WebExtensionSQLiteStore::close):
(WebKit::WebExtensionSQLiteStore::deleteDatabase):
(WebKit::WebExtensionSQLiteStore::vacuum):
(WebKit::WebExtensionSQLiteStore::openDatabaseIfNecessary):
(WebKit::WebExtensionSQLiteStore::openDatabase):
(WebKit::WebExtensionSQLiteStore::isDatabaseOpen):
(WebKit::WebExtensionSQLiteStore::deleteDatabaseFileAtURL):
(WebKit::WebExtensionSQLiteStore::deleteDatabaseIfEmpty):
(WebKit::WebExtensionSQLiteStore::handleSchemaVersioning):
(WebKit::WebExtensionSQLiteStore::migrateToCurrentSchemaVersionIfNeeded):
(WebKit::WebExtensionSQLiteStore::databaseSchemaVersion):
(WebKit::WebExtensionSQLiteStore::setDatabaseSchemaVersion):
(WebKit::WebExtensionSQLiteStore::savepointNameFromUUID):
(WebKit::WebExtensionSQLiteStore::createSavepoint):
(WebKit::WebExtensionSQLiteStore::commitSavepoint):
(WebKit::WebExtensionSQLiteStore::rollbackToSavepoint):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStore.h: Added.
(WebKit::WebExtensionSQLiteStore::~WebExtensionSQLiteStore):
(WebKit::WebExtensionSQLiteStore::useInMemoryDatabase):
(WebKit::WebExtensionSQLiteStore::queue):
(WebKit::WebExtensionSQLiteStore::database):
(WebKit::WebExtensionSQLiteStore::uniqueIdentifier):
(WebKit::WebExtensionSQLiteStore::lastErrorMessage):
(WebKit::WebExtensionSQLiteStore::directory):
* Source/WebKit/Sources.txt:
* Source/WebKit/UnifiedSources-output.xcfilelist:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/Scripts/webkitpy/style/checker.py:

Canonical link: <a href="https://commits.webkit.org/289200@main">https://commits.webkit.org/289200@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/689b034fff04992619ded30d39f5f7f96727b719

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/85342 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/5077 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/39773 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/90470 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/36384 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/87431 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/5166 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/13054 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/66343 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/24158 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/88388 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/3937 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/77519 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/46625 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/84749 "webkitpy-tests (failure)") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/3819 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/31780 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/35452 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/74540 "1 api test failed or timed out") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/32618 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/91960 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/12689 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/9277 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/74924 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/12918 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/73356 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/74043 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/18423 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/16854 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/4710 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13362 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/12632 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/18097 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/12462 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/15955 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/14213 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.h:
```diff
@@ -0,0 +1,82 @@
+};
+
+class WebExtensionSQLiteRowEnumerator : public RefCounted<WebExtensionSQLiteRowEnumerator> {
+public:
```
```suggestion
class WebExtensionSQLiteRowEnumerator : public RefCounted<WebExtensionSQLiteRowEnumerator> {
    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteRowEnumerator);
    WTF_MAKE_TZONE_ALLOCATED(WebExtensionSQLiteRowEnumerator);

public:
```

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.h:
```diff
@@ -0,0 +1,82 @@
+
+    RefPtr<WebExtensionSQLiteRow> next();
+    Ref<WebExtensionSQLiteStatement> statement() { return m_statement; };
+private:
```
```suggestion
    Ref<WebExtensionSQLiteStatement> statement() { return m_statement; };

private:
```

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.h:
```diff
@@ -0,0 +1,103 @@
+
+    friend class WebExtensionSQLiteStatement;
+    friend class WebExtensionSQLiteStore;
+public:
```
```suggestion
    friend class WebExtensionSQLiteStatement;
    friend class WebExtensionSQLiteStore;

public:
```

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.h:
```diff
@@ -0,0 +1,103 @@
+class WebExtensionSQLiteStore;
+
+class WebExtensionSQLiteDatabase final : public RefCounted<WebExtensionSQLiteDatabase> {
+    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteDatabase);
```
```suggestion
    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteDatabase);
    WTF_MAKE_TZONE_ALLOCATED(WebExtensionSQLiteDatabase);
```

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.h:
```diff
@@ -0,0 +1,82 @@
+namespace WebKit {
+
+class WebExtensionSQLiteRow : public RefCounted<WebExtensionSQLiteRow> {
+    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteRow);
```
```suggestion
    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteRow);
    WebExtensionSQLiteDatabase(WebExtensionSQLiteRow);
```

ItsJamie9494 replied:
I'm not totally sure why adding WebExtensionSQLiteDatabase(WebExtensionSQLiteRow) is needed here?

xeenon replied:
Oops, copy-paste error.
```suggestion
    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteRow);
    WTF_MAKE_TZONE_ALLOCATED(WebExtensionSQLiteRow);
```

xeenon commented on Source/WebKit/Shared/Extensions/WebExtensionSQLiteStatement.h:
```diff
@@ -0,0 +1,87 @@
+class WebExtensionSQLiteRowEnumerator;
+
+class WebExtensionSQLiteStatement : public RefCounted<WebExtensionSQLiteStatement> {
+    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteStatement);
```
```suggestion
    WTF_MAKE_NONCOPYABLE(WebExtensionSQLiteStatement);
    WTF_MAKE_TZONE_ALLOCATED(WebExtensionSQLiteStatement);
```

xeenon reviewed (approved):
Looks good!

ItsJamie9494 committed:
Create a C++ implementation of WebExtensionSQLite classes
https://webkit.org/b/285893

Reviewed by Timothy Hatcher.

This adds a new WebExtensionSQLite implementation, with the Database, Helpers, Row, Statements, and Store classes.
Currently, there is no usage of this implementation, as StorageSQLiteStore, RegisteredScriptsSQLiteStore, and DeclarativeNetRequestSQLiteStore should be ported separately.

* Source/WebKit/Scripts/generate-unified-sources.sh:
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.cpp: Added.
(WebExtensionSQLiteDatabase::WebExtensionSQLiteDatabase):
(WebExtensionSQLiteDatabase::assertQueue):
(WebExtensionSQLiteDatabase::close):
(WebExtensionSQLiteDatabase::reportErrorWithCode):
(WebExtensionSQLiteDatabase::errorWithSQLiteErrorCode):
(WebExtensionSQLiteDatabase::enableWAL):
(WebExtensionSQLiteDatabase::openWithAccessType):
(WebExtensionSQLiteDatabase::inMemoryDatabaseURL):
(WebExtensionSQLiteDatabase::privateOnDiskDatabaseURL):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatabase.h: Added.
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteDatatypeTraits.h: Added.
(WebKit::WebExtensionSQLiteDatatypeTraits<int>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<int>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<int64_t>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<int64_t>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<double>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<double>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<String>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<String>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<RefPtr<API::Data>>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<RefPtr<API::Data>>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<std::nullptr_t>::fetch):
(WebKit::WebExtensionSQLiteDatatypeTraits<std::nullptr_t>::bind):
(WebKit::WebExtensionSQLiteDatatypeTraits<decltype):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteHelpers.h: Added.
(WebKit::SQLiteIsExecutionError):
(WebKit::SQLiteDatabaseExecuteAndReturnIntError):
(WebKit::SQLiteStatementBindAllParameters):
(WebKit::SQLiteDatabaseExecute):
(WebKit::SQLiteDatabaseExecuteAndReturnError):
(WebKit::SQLiteDatabaseFetch):
(WebKit::SQLiteStatementBindParameter):
(WebKit::SQLiteStatementBindParameter<unsigned>):
(WebKit::SQLiteStatementBindParameter<uint64_t>):
(WebKit::SQLiteStatementBindParameter<long>):
(WebKit::SQLiteStatementBindParameter<bool>):
(WebKit::SQLiteStatementBindTupleParameters):
(WebKit::SQLiteIteratorBlock::SQLiteIteratorBlock):
(WebKit::SQLiteIteratorBlock::callBlockWithAllColumns):
(WebKit::StatementCallBlockWithAllColumns):
(WebKit::WBSStatementFetchColumnsInTuple):
(WebKit::SQLiteStatementBindOrStep):
(WebKit::SQLiteStatementBind):
(WebKit::SQLiteDatabaseEnumerate):
(WebKit::SQLiteDatabaseEnumerateRows):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.cpp: Added.
(WebKit::WebExtensionSQLiteRow::WebExtensionSQLiteRow):
(WebKit::WebExtensionSQLiteRow::getString):
(WebKit::WebExtensionSQLiteRow::getInt):
(WebKit::WebExtensionSQLiteRow::getInt64):
(WebKit::WebExtensionSQLiteRow::getDouble):
(WebKit::WebExtensionSQLiteRow::getBool):
(WebKit::WebExtensionSQLiteRow::getData):
(WebKit::WebExtensionSQLiteRow::isNullAtIndex):
(WebKit::WebExtensionSQLiteRowEnumerator::WebExtensionSQLiteRowEnumerator):
(WebKit::WebExtensionSQLiteRowEnumerator::next):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteRow.h: Added.
(WebKit::WebExtensionSQLiteRow::create):
(WebKit::WebExtensionSQLiteRowEnumerator::create):
(WebKit::WebExtensionSQLiteRowEnumerator::statement):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStatement.cpp: Added.
(WebKit::WebExtensionSQLiteStatement::WebExtensionSQLiteStatement):
(WebKit::WebExtensionSQLiteStatement::~WebExtensionSQLiteStatement):
(WebKit::WebExtensionSQLiteStatement::execute):
(WebKit::WebExtensionSQLiteStatement::fetch):
(WebKit::WebExtensionSQLiteStatement::fetchWithEnumerationCallback):
(WebKit::WebExtensionSQLiteStatement::reset):
(WebKit::WebExtensionSQLiteStatement::invalidate):
(WebKit::WebExtensionSQLiteStatement::bind):
(WebKit::WebExtensionSQLiteStatement::columnNamesToIndicies):
(WebKit::WebExtensionSQLiteStatement::columnNames):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStatement.h: Added.
(WebKit::WebExtensionSQLiteStatement::create):
(WebKit::WebExtensionSQLiteStatement::database):
(WebKit::WebExtensionSQLiteStatement::handle):
(WebKit::WebExtensionSQLiteStatement::isValid):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStore.cpp: Added.
(WebKit::WebExtensionSQLiteStore::WebExtensionSQLiteStore):
(WebKit::WebExtensionSQLiteStore::close):
(WebKit::WebExtensionSQLiteStore::deleteDatabase):
(WebKit::WebExtensionSQLiteStore::vacuum):
(WebKit::WebExtensionSQLiteStore::openDatabaseIfNecessary):
(WebKit::WebExtensionSQLiteStore::openDatabase):
(WebKit::WebExtensionSQLiteStore::isDatabaseOpen):
(WebKit::WebExtensionSQLiteStore::deleteDatabaseFileAtURL):
(WebKit::WebExtensionSQLiteStore::deleteDatabaseIfEmpty):
(WebKit::WebExtensionSQLiteStore::handleSchemaVersioning):
(WebKit::WebExtensionSQLiteStore::migrateToCurrentSchemaVersionIfNeeded):
(WebKit::WebExtensionSQLiteStore::databaseSchemaVersion):
(WebKit::WebExtensionSQLiteStore::setDatabaseSchemaVersion):
(WebKit::WebExtensionSQLiteStore::savepointNameFromUUID):
(WebKit::WebExtensionSQLiteStore::createSavepoint):
(WebKit::WebExtensionSQLiteStore::commitSavepoint):
(WebKit::WebExtensionSQLiteStore::rollbackToSavepoint):
* Source/WebKit/Shared/Extensions/WebExtensionSQLiteStore.h: Added.
(WebKit::WebExtensionSQLiteStore::~WebExtensionSQLiteStore):
(WebKit::WebExtensionSQLiteStore::useInMemoryDatabase):
(WebKit::WebExtensionSQLiteStore::queue):
(WebKit::WebExtensionSQLiteStore::database):
(WebKit::WebExtensionSQLiteStore::uniqueIdentifier):
(WebKit::WebExtensionSQLiteStore::lastErrorMessage):
(WebKit::WebExtensionSQLiteStore::directory):
* Source/WebKit/Sources.txt:
* Source/WebKit/UnifiedSources-output.xcfilelist:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/Scripts/webkitpy/style/checker.py:

Canonical link: https://commits.webkit.org/289200@main
